### PR TITLE
Add support for access token auth; add mutateAlertRule; add CLI

### PR DESCRIPTION
**Pull request by erkangz**
Add the following to `j1client`
- ability to use an API access token directly instead of username/password
- `mutateAlertRule` 

Add `j1cli` and update `README`

### CODE REVIEW

**Comment by erichs** [comment]
Nice!

**Comment by erkangz** [comment]
i'll add relationships support in a separate PR

File: src/j1cli.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  let data;
+
+  if (!program.query || program.query === '') {
+    if (!program.entity && !program.relationship && !program.alert) {
+      error.fatal('Must specify a query (using -q|--query) or an entity/relationship/rule operation (--entity or --relationship or --alert)', EUSAGEERROR);
+    } else if (!program.operation || program.operation === '') {
+      error.fatal('Must specify command action for entity or rule operation (-c|--command <create|update|delete>)', EUSAGEERROR);
+    } else if (!program.file || program.file === '') {
+      error.fatal('Must specify input JSON file with -f|--file)', EUSAGEERROR);
+    } else if (!fs.existsSync(program.file)) {
```
Comment (austinkelleher) [inline src/j1cli.js]: This is already an async function. You may as well `promisify` the `fs.exists` function and just `await` on it.

File: src/j1cli.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      await gatherPassword();
+    }
+  }
+
+  return data;
+}
+
+function jParse (file) {
+  let data;
+  try { data = JSON.parse(fs.readFileSync(file, 'utf8')); } catch (err) { return null; }
```
Comment (austinkelleher) [inline src/j1cli.js]: Similar to the above, you can `promisify` the `fs.readFile` function and `await` on it.

File: src/j1cli.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+async function mutateAlertRule(j1Client, rule, update) {
+  const res = await j1Client.mutateAlertRule(rule, update);
+  const ruleId = res.id;
+  return ruleId;
+}
+
+async function createAlerts(j1Client, rules) {
+  const ruleIds = [];
+  for (const r of rules) {
+    const ruleId = await mutateAlertRule(j1Client, r, false);
```
Comment (austinkelleher) [inline src/j1cli.js]: You could use `Promise.all` here instead.

File: src/j1cli.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    ruleIds.push(ruleId);
+    console.log(`Created alert rule ${ruleId}.`);
+  }
+}
+
+async function updateAlerts(j1Client, rules) {
+  const ruleIds = [];
+  for (const r of rules) {
+    const ruleId = await mutateAlertRule(j1Client, r, true);
```
Comment (austinkelleher) [inline src/j1cli.js]: You could use `Promise.all` here instead.

**Pull request merged**